plugins/statistics.koplugin/main.lua

```diff
@@ -234,28 +248,112 @@ function getDatesForBookOldFormat(book)
     return generateReadBooksTable(book.title, dates)
 end
 
-function getDatesForBook(book)
+-- sdays -> number of days to show
+-- ptype -> daily - show daily without weekday name
+--          daily_weekday - show daily with weekday name
+--          weekly - show weekly
+--          monthly - show monthly
+function getDatesFromAll(sdays, ptype)
     local dates = {}
+    local sorted_tbl = {}
```

Comment:
you are already assigning a new table to this variable in the for loop, so there is no need to create an empty table here :)